packages/frontend/src/common/hooks/useValidatedTimeInput.ts

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+
+interface useValidatedTimeInputOptions {
+  formatLength?: number;
+  minValue?: number;
+  maxValue?: number;
+  initialValue?: number;
+}
+export function useValidatedTimeInput(options: useValidatedTimeInputOptions | undefined = {}) {
+  const [value, setValue] = useState(options.initialValue ?? 0);
+  const [formatedValue, setFormatedValue] = useState(
```

Comment:
As I understood, the formattedValue will only change, when the value changes. So you have a direct coupling between value and formattedValue. In this case, we never want to manage two states. `formattedValue` can simply be calculated from `value`. How do we do this?

We just calculated `formattedValue` on each render.

```
  const formattedValue = addLeadingZeros(value);
```
We can also remove `setFormatedValue(addLeadingZeros(value));` in the change function, since this value is calculated automatically as soon as the value changes now.